shell/browser/api/electron_api_web_contents.cc

```diff
@@ -684,6 +688,15 @@ void WebContents::WebContentsCreatedWithFullParams(
   tracker->referrer = params.referrer.To<content::Referrer>();
   tracker->raw_features = params.raw_features;
   tracker->body = params.body;
+
+  v8::HandleScope scope(isolate());
+  v8::Local<v8::Value> val =
+      gin::ConvertToV8(isolate(), pending_child_web_preferences_);
+  gin_helper::Dictionary dict;
+  gin::ConvertFromV8(isolate(), val, &dict);
+  pending_child_web_preferences_.DictEmpty();
+
+  new WebContentsPreferences(new_contents, dict);
```

Comment:
I would very much appreciate a less insane way to do this.

Reply:
@loc You can temporarily put the `v8::Value` into a `v8::Global`, that allows it to exist for longer than the stack it was allocated on.